src/v/cluster/rm_stm.cc

```diff
@@ -79,17 +79,28 @@ static model::record_batch make_prepare_batch(rm_stm::prepare_marker record) {
 static model::record_batch make_tx_control_batch(
   model::producer_identity pid, model::control_record_type crt) {
     iobuf key;
-    kafka::response_writer w(key);
-    w.write(model::current_control_record_version());
-    w.write(static_cast<int16_t>(crt));
+    kafka::response_writer kw(key);
+    kw.write(model::current_control_record_version());
+    kw.write(static_cast<int16_t>(crt));
+
+    std::optional<iobuf> value_opt = std::nullopt;
+    if (crt == model::control_record_type::tx_commit) {
```

Comment:
Do we need to do the same thing for `model::control_record_type::tx_abort`?
In sarama they assert for `ControlRecordAbort` and `ControlRecordCommit`

Reply:
Awesome catch, thanks! Updated

Reply:
should this be an `allow-list` rather than a `deny-list` in the sense that if we do the opposite only allow known upstream control batches, then as we continue to expand it will always work? just a thought.

Reply:
rm_stm works only with tx control batches and if we decide to expand it to support other control batches (which are not in the kafka protocol yet) we anyway we need to redesign rm_stm so it's safe to leave it as is